stock_picking_batch_extended/report/batch_report.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2018 Tecnativa - Carlos Dauden
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import logging
+
+from odoo import api, fields, models
+
+
+_logger = logging.getLogger(__name__)
+
+
+class ReportPrintBatchPicking(models.AbstractModel):
+    _name = 'report.stock_picking_batch_extended.report_batch_picking'
+    _description = 'Report for Batch Picking'
+
+    @api.model
+    def key_level_0(self, operation):
+        return operation.location_id.id, operation.location_dest_id.id
+
+    @api.model
+    def key_level_1(self, operation):
+        return operation.product_id.id
+
+    @api.model
+    def new_level_0(self, operation):
+        level_0_name = u'{} \u21E8 {}'.format(
+            operation.location_id.name_get()[0][1],
+            operation.location_dest_id.name_get()[0][1])
+        return {
+            'name': level_0_name,
+            'location': operation.location_id,
+            'location_dest': operation.location_dest_id,
+            'l1_items': {},
+        }
+
+    @api.model
+    def new_level_1(self, operation):
+        return {
+            'product': operation.product_id,
+            'product_qty': operation.product_qty or operation.qty_done,
```

Comment:
I would have written :

```suggestion
            'product_qty': not float_is_zero(operation.product_qty) or operation.qty_done,
```

Reply:
I don't understand this...

Reply:
Do you mean @rousseldenis's comment or the original code?

Reply:
the change supposes to say that 
if product_qty = 0.00000001 and precision is set 0.001, product_qty is equal to 0, 
if use just (not product_qty) than 0.00000001 will be considered as not 0